geoh5py/groups/interpretation_section.py

```diff
@@ -0,0 +1,312 @@
+#  Copyright (c) 2024 Mira Geoscience Ltd.
+#
+#  This file is part of geoh5py.
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import TYPE_CHECKING, Any
+from uuid import UUID
+
+from pydantic import BaseModel, Field
+
+from ..shared.utils import str2uuid, to_tuple
+from .base import Group
+
+
+if TYPE_CHECKING:  # pragma: no cover
+    from ..objects import Curve, ObjectBase, Slicer
+
+
+class InterpretationSectionParams(BaseModel):
+    """
+    The class to store the interpretation section.
+
+    :param normal_x: The normal x component.
+    :param normal_y: The normal y component.
+    :param normal_z: The normal z component.
+    :param position_x: The position x component.
+    :param position_y: The position y component.
+    :param position_z: The position z component.
+    """
+
+    normal_x: float = Field(alias="Normal X")
+    normal_y: float = Field(alias="Normal Y")
+    normal_z: float = Field(alias="Normal Z")
+    position_x: float = Field(alias="Position X")
+    position_y: float = Field(alias="Position Y")
+    position_z: float = Field(alias="Position Z")
+
+    @classmethod
+    def create_section(
+        cls,
+        params: dict | InterpretationSectionParams,
+    ) -> InterpretationSectionParams:
+        """
+        Verify/Create the interpretation section from a mapper.
+
+        :param params: The parameters to verify.
+
+        :return: An instance of the class.
+        """
+        if isinstance(params, Mapping):
```

Comment:
Fancy new type `Mapping`. Why is that?

Reply:
a Mapping is a mapper containing key and values.
Just to be more fancy than "dict. but it could be "dict" too

Reply:
Sounds special. I ll let you have it ;)